kornia/core/module.py

```diff
@@ -0,0 +1,275 @@
+
+        Args:
+            input_names_to_handle: List of input names to convert, if None, handle all inputs.
+            output_type: Desired output type ('tensor', 'numpy', or 'pil').
+
+        Returns:
+            Callable: Decorated function with converted input and output types.
+        """
+
+        def decorator(func: Callable[[Any], Any]) -> Callable[[Any], Any]:
+            @wraps(func)
+            def wrapper(*args: Any, **kwargs: Any) -> Union[Any, List[Any]]:
+                # If input_names_to_handle is None, handle all inputs
+                if input_names_to_handle is None:
+                    # Convert all args to tensors
+                    args = tuple(self.to_tensor(arg) if self._is_valid_arg(arg) else arg for arg in args)
+                    # Convert all kwargs to tensors
+                    kwargs = {k: self.to_tensor(v) if self._is_valid_arg(v) else v for k, v in kwargs.items()}
+                else:
+                    # Convert specified args to tensors
+                    args = list(args)  # type:ignore
+                    for i, (arg, name) in enumerate(zip(args, func.__code__.co_varnames)):
+                        if name in input_names_to_handle:
+                            args[i] = self.to_tensor(arg)  # type:ignore
+                    # Convert specified kwargs to tensors
+                    for name, value in kwargs.items():
+                        if name in input_names_to_handle:
+                            kwargs[name] = self.to_tensor(value)
+
+                # Call the actual forward method
+                tensor_outputs = func(*args, **kwargs)
+
+                if not isinstance(tensor_outputs, (tuple,)):
+                    tensor_outputs = (tensor_outputs,)
+
+                # Convert outputs to the desired type
+                outputs = []
+                for output in tensor_outputs:
+                    if output_type == "tensor":
+                        outputs.append(output)
+                    elif output_type == "numpy":
+                        outputs.append(self.to_numpy(output))
+                    elif output_type == "pil":
+                        outputs.append(self.to_pil(output))
+                    else:
+                        raise ValueError("Output type not supported. Choose from 'tensor', 'numpy', or 'pil'.")
+
+                return outputs if len(outputs) > 1 else outputs[0]
+
+            return wrapper
+
+        return decorator
+
+    def _is_valid_arg(self, arg: Any) -> bool:
+        """Check if the argument is a valid type for conversion.
+
+        Args:
+            arg: The argument to check.
+
+        Returns:
+            bool: True if valid, False otherwise.
+        """
+        if isinstance(arg, (str,)) and os.path.exists(arg):
+            return True
+        if isinstance(arg, (Tensor,)):
+            return True
+        # Make sure that the numpy and PIL are not necessarily needed to be imported.
```

Comment:
i think this and the Image.Image access to check the type will always try to lazyload the modules no? If yes, it'll always crash if we do not have the deps

Reply:
If the function returns within the first str and tensor type checks, it will not come to execute the attribute access. Then it will not crash.

One drawback is that those lines will never pass the static type check since those modules are loaded in the runtime.